### PR TITLE
Fix name regex

### LINKED ISSUE #101: [PE-D] Format of student names is not checked.
Name of student can be purely numerical.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: lawwm/ped#2

### LINKED ISSUE #122: [PE-D] Hyphenated last names are not allowed.
For example, Joseph Gordon-Levitt is a legitimate name, but the program rejects it.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: jltham/ped#11

### LINKED ISSUE #137: [PE-D] Names containing exclusively numbers are allowed
Feature tested: add

Steps to reproduce: add n/222 tg/CS2100 G08 e/e0123456@u.nus.edu

Issue: Having name be made up of exclusively numbers doesn't seem realistic

Suggestion: There must be some words with characters, or have offsprings use letters to represent numbers instead

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: jinnhl/ped#4

### PR DESCRIPTION
**Pull request by jaysmyname**
Change name to allow hyphens and apostrophes
Numbers are allowed as long as they are at the end
Restrict name character limit to 100

Resolve #101
Resolve #122
Resolve #137

### CODE REVIEW

File: src/main/java/seedu/address/model/student/Name.java [hunk 1]
```diff
@@ -10,13 +10,14 @@
 public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain letters, and can contain a single space, apostrophe or hyphen surrounded by "
+                    + "letters. Names cannot be blank and cannot be longer than 100 characters in total.";
```
Comment (LowJiaHao99) [inline src/main/java/seedu/address/model/student/Name.java:14]: Add `Can end with numbers` to sync with user guide

**Review by LowJiaHao99 — CHANGES_REQUESTED** [review]
Just 1 small change needed, as well as a clarifying question. Once the small issue is resolved, LGTM.

File: src/main/java/seedu/address/model/student/Name.java [hunk 2]
```diff
@@ -10,13 +10,14 @@
...
 
     /*
      * The first character of the address must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    public static final String VALIDATION_REGEX = "[A-Za-z]([ '-]?[A-Za-z])*( \\d+)?";
```
[the remark this answers is not in the document]
Comment (jaysmyname) [reply src/main/java/seedu/address/model/student/Name.java:20]: Yes. See NameTest.java line 39

File: src/main/java/seedu/address/model/student/Name.java [hunk 3]
```diff
@@ -10,13 +10,15 @@
 public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain letters, and can contain a single space, apostrophe or hyphen surrounded by "
+                    + "letters. Names can end with numbers. Names cannot be blank and cannot be longer than "
```
Comment (vanessaxuuan) [inline src/main/java/seedu/address/model/student/Name.java:14]: Could edit to "Names must start with a letter but can end with numbers"?

**Review by LowJiaHao99 — APPROVED** [review]

**Comment by LowJiaHao99** [comment]
LGTM

**Pull request merged**